lib/db/jdb-internal.c

```diff
@@ -213,6 +213,8 @@ j_db_internal_schema_get(JDBSchema* j_db_schema, JBatch* batch, GError** error)
 
 	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
 
+	j_operation_cache_flush(); // flush in-flight batches
```

Comment:
Does it make sense to flush implicitly (within the operation cache or batch) when a read operation is performed? We probably also only need to flush once the operation is actually executed, not when it is scheduled.